### ISSUE TITLE
Introduce cop for redundant ternary condition

### ISSUE THREAD

**Issue by Drenmi** [bbatsov/rubocop#4130]
I'm working on fixing up a legacy project, and I keep seeing this over and over:

```ruby
foo == bar ? true : false
```

We should have a cop to detect this.

We could also detect the inverse:

```ruby
foo == bar ? false : true
```

and suggest:

```ruby
foo != bar
```

This should also work for this construct:

```ruby
foo? ? false : true
```

which can be replaced with:

```ruby
!foo?
```

I'm not sure how common this is for regular `if`-statements, but could also be extended to cover those.

--------

## Expected behavior

RuboCop registers an offense for:

```ruby
condition ? true : false
```

## Actual behavior

RuboCop does not register an offense.

## Steps to reproduce the problem

Run RuboCop on:

```ruby
condition ? true : false
```

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.47.1 (using Parser 2.3.3.1, running on ruby 2.3.3 x86_64-darwin15)
```

**Comment by bbatsov** [comment]
Great suggestion!

**Comment by Drenmi** [comment]
Another case where a ternary is redundant:

```ruby
foo = bar? ? baz : nil
```

can better be written as:

```ruby
foo = baz if bar?
```

and the inverse:

```ruby
foo = bar? ? nil : baz
```

as:

```ruby
foo = baz unless bar?
```

**Comment by petehamilton** [comment]
@Drenmi @bbatsov I'd be up for taking a stab at this one -would you be cool with me putting something together and submitting it for your review/feedback?

**Comment by Drenmi** [comment]
That would be awesome @petehamilton. 🙂 

Depending on your personal time constraints, I would recommend building the basic cases first, then adding the remaining ones in subsequent PRs. We see a lot of PRs fizzle, unfortunately. 🙂 

Feel free to ping me if you have any questions.

**Comment by petehamilton** [comment]
@Drenmi absolutely - that was my plan (especially having worked on rubocop relatively little in custom cops).

My rough plan was initially:
- Spot `x ? true : false` and `x ? false : true` offenses
- Auto-correct those offenses

Then once I've confirmed I'm on the right track, potentially:
- Do the same for `if/else`
- Do the same for `foo = bar? ? baz : nil`

Sound sensible?

**Comment by petehamilton** [comment]
@Drenmi do you have an opinion on valid conditionals to apply this to?

We can't guarantee that a method returns a boolean so in the below example we'd be changing behaviour. We presumably want to limit this to `== === != < > <= >= <=>` and `foo?` convention methods?

```
def foo(x)
  x ? true : false
end

def new_foo(x)
  x
end
```

**Comment by petehamilton** [comment]
@Drenmi had a but of time at lunch so took an initial stab here: https://github.com/bbatsov/rubocop/compare/master...petehamilton:implement-redundant-ternary

Would be great to get some initial feedback before I continue.

**Comment by Drenmi** [comment]
> We can't guarantee that a method returns a boolean so in the below example we'd be changing behaviour. We presumably want to therefore limit this to == === != < > <= >= <=> and foo? convention methods?

I think this is reasonable for a start. I personally think it is still a bad idea to use a ternary like this, because it basically works around Ruby's concept of truthiness and falsiness, but making the cop more far-reaching by default will likely be controversial.

> Would be great to get some initial feedback before I continue.

If you open a PR, I can comment directly on the code for you. 🙂

**Comment by petehamilton** [comment]
@drenmi was holding off as I didn't want to have an in-flux PR hanging around not ready for review but if that's better for you can-do!

https://github.com/bbatsov/rubocop/pull/4453

**Comment by Drenmi** [comment]
Looking good @petehamilton! I left some comments in the PR. 🙂
